go/pkg/cosmos/history.go

```diff
@@ -19,36 +20,54 @@ type TxState struct {
 	lastID  string
 	page    int
 	query   string
-	txs     []client.TxSearchResponseResultTxs
+	txs     []TxSearchResponseResultTxs
 }
 
 // History stores state for multiple query sources to complete a paginated request
 type History struct {
-	ctx              context.Context
-	cursor           *Cursor
-	encoding         *params.EncodingConfig
-	pageSize         int
-	receive          *TxState
-	send             *TxState
-	tendermintClient *client.APIClient
+	ctx        context.Context
+	cursor     *Cursor
+	encoding   *params.EncodingConfig
+	pageSize   int
+	receive    *TxState
+	send       *TxState
+	tendermint *resty.Client
 }
 
-func (h *History) doRequest(txState *TxState) (*client.TxSearchResponse, error) {
+func (h *History) doRequest(txState *TxState) (*TxSearchResponse, error) {
 	for {
-		res, httpRes, err := h.tendermintClient.InfoApi.TxSearch(h.ctx).PerPage(STATIC_PAGE_SIZE).OrderBy("\"desc\"").Query(txState.query).Page(int32(txState.page)).Execute()
-		if err != nil {
+		var res *TxSearchResponse
```

Comment:
The commit suggestion is unfortunately not working for the code block I would like, but we can correctly type the error response so we can leverage the `SetError()` and reduce the need for the status code check and subsequent unmarshal:
```
		var res *TxSearchResponse
		var e *struct {
			Jsonrpc string `json:"jsonrpc"`
			Id      int32  `json:"id"`
			Error   struct {
				Data string `json:"data"`
			} `json:"error"`
		}

		queryParams := map[string]string{
			"per_page": strconv.Itoa(int(STATIC_PAGE_SIZE)),
			"order_by": "\"desc\"",
			"query":    txState.query,
			"page":     strconv.Itoa(txState.page),
		}

		_, err := h.tendermint.R().SetResult(&res).SetError(&e).SetQueryParams(queryParams).Get("/tx_search")
		if err != nil {
			return nil, errors.Wrap(err, "failed to do request")
		}

		if e != nil {
			// error is returned if page is out of range past page 1, mark as no more transactions
			if strings.Contains(e.Error.Data, "page should be within") {
				txState.hasMore = false
				return &TxSearchResponse{}, nil
			}

			return nil, errors.Errorf("failed to do request: %s", e.Error.Data)
		}
```

Reply:
I also noticed your use of `fmt.Sprint` and have been using `strconv.Itoa` elsewhere and would like to standardize on that style.